### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by imkiva**
- Reorganize problems by their type
- Less visitors
- Delete really unused classes & methods

### CODE REVIEW

File: base/src/main/java/org/aya/resolve/error/PrimResolveError.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  @Override default @NotNull Stage stage() {return Stage.RESOLVE;}
+  @Override default @NotNull Severity level() {return Severity.ERROR;}
+
+  record UnknownPrim(
+    @Override @NotNull SourcePos sourcePos,
+    @NotNull String name
+  ) implements PrimResolveError {
+    @Override public @NotNull Doc describe(@NotNull DistillerOptions options) {
+      return Doc.sep(
+        Doc.plain("Unknown primitive"),
```
Comment (ice1000) [inline base/src/main/java/org/aya/resolve/error/PrimResolveError.java:24]: English

**Review by ice1000 — COMMENTED** [review]
Excellent!

Are we eventually going to use the cancellation indicators? I still kinda think they're useful. Maybe it's because we didn't have very long code at this point.

**Comment by imkiva** [comment]
> Excellent!
> 
> Are we eventually going to use the cancellation indicators? I still kinda think they're useful. Maybe it's because we didn't have very long code at this point.

I don't know either... but they are more useful as UI actions rather than here (since you can always use ctrl-c for cmdline apps).

File: base/src/main/java/org/aya/resolve/error/PrimResolveError.java (hunk 1 above)
Comment (imkiva) [reply base/src/main/java/org/aya/resolve/error/PrimResolveError.java]: nice catch~

**Comment by ice1000** [comment]
@imkiva Yes but what about LSP

**Comment by imkiva** [comment]
> @imkiva Yes but what about LSP

It applies the way JVM terminates a completable future.

**Comment by ice1000** [comment]
bors r_

**Comment by imkiva** [comment]
bors cancel

**Comment by imkiva** [comment]
bors merge

**Pull request merged**